lib/umwelt/parser.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Umwelt.Parser do
+  @moduledoc "Extracts metadata from AST"
+
+  alias Umwelt.Parser
+
+  def read_ast({:error, msg}), do: {:error, msg}
+
+  def read_ast({:ok, code}),
+    do: Code.string_to_quoted(code)
+
+  def parse({:ok, ast}),
+    do: parse(ast, []) |> index()
+
+  def parse({_, _, _} = ast, aliases),
+    do: Parser.Triple.parse(ast, aliases)
+
+  def parse({_, _} = ast, aliases),
+    do: Parser.Tuple.parse(ast, aliases)
+
+  def parse(ast, aliases) when is_list(ast),
+    do: ast |> Enum.map(&parse(&1, aliases))
```

Comment:
```suggestion
    do: Enum.map(ast, &parse(&1, aliases))
```
Look, there are community standards, which are:

- no pipe for less than two piped applications, it’s not a pipe, it’s a regular call then
- start the pipe with the variable

You might not follow those, but it makes it harder to read your code for all the active community members :)